src/modules-by-example.adoc

```diff
@@ -319,3 +320,472 @@ Here again, Ion 1.1 enables a new technique: we can prepend new symbols to the e
+
+While the signature of `point` is now implicit in the signature of `scatterplot`, and while the
+macro expander will invoke `point` while expanding `scatterplot`, neither `point` nor
+the module containing it is in scope within the document:
+
+[{nrm}]
+----
+(:point 25 10)   ⇒ **error**: no installed module exports a macro named 'point'.
+(:geo:point 2 1) ⇒ **error**: no module named 'geo' is installed.
+----
+
+In particular, `geo` is not in the encoding environment's available modules, since it wasn't
+imported into it:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [(*import* chart "com.example.charts" 1)],
+  install_macros: [chart, geo]
+}
+  ⇒ **error**: no module named 'geo' is available for installation.
+----
+
+When the Ion implementation loads the 'chart' module, it will transitively load the geometry
+module as well, but the import of `com.example.geometry` by `com.example.charts` is
+_private_ to the importer.
+
+You can do similar things within an encoding directive:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*module* geo
+      (*macro* point [(int x), (int y)]
+        {x: x, y: y})
+      (*macro* line  [(point a), (point b)]
+        [a, b]))
+    (*module* chart
+      (*import* geo)                                 // <1>
+      (*macro* scatterplot [(point\... points)]
+        [points]))
+  ],
+  install_macros: [ chart ]                        // <2>
+}
+----
+
+<1> Importing `geo` makes its macros accessible within `chart`.
+<2> The `geo` module is not installed into the encoding environment, so its macros are not
+accessible in the document body.
+
+
+=== Macro Aliases
+
+We've seen how to resolve an ambiguous macro name by using qualified references.  Another approach
+is to give new names to existing macros.  Suppose we want to add a 3d chart to our module, so we
+import both the 2d and 3d modules:
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*module* chart
+      (*import* geo "com.example.geometry" 1)
+      (*import* g3d "com.example.graphics.3d" 1)
+      (*macro* scatterplot [(point\... points)]
+
+  ⇒ **error**: 'point' is ambiguous, exported by 'geo' and 'g3d'.
+----
+
+The most direct way to fix this is to use a qualified reference.  We've seen this used in
+E-expressions like `(:geo:point 17 28)`, but now we need it in a signature where the special
+smile syntax does not apply.  Instead, use a quoted symbol:
+
+[{nrm}]
+----
+      (*macro* scatterplot [(':geo:point' \... points)]
+        [points]))
+----
+
+That has the intended effect of keeping `scatterplot` using 2D points, but it's somewhat awkward.
+A more ergonomic approach is to introduce an alias to disambugate:
+
+
+[{nrm}]
+----
+  modules: [
+    (*module* chart
+      (*import* geo "com.example.geometry" 1)
+      (*import* g3d "com.example.graphics.3d" 1)
+      (*alias* point2 ':geo:point')               // <1>
+      (*macro* scatterplot [(point2 \... points)]  // <2>
```

Comment:
Yes, you could write `(':point2' \... points)` in the signature, since that colon forces resolution among macro/alias declarations in this module.

But I kinda want to avoid getting into all the rules here.

PS I didn't think this was clever, its kind a the point of the example...